python/tvm/_ffi/_cython/ndarray.pxi

```diff
@@ -73,8 +73,15 @@ cdef class NDArrayBase:
         return pycapsule.PyCapsule_New(dltensor, _c_str_dltensor, _c_dlpack_deleter)
 
 
-cdef c_make_array(void* chandle, is_view):
-    ret = _CLASS_NDARRAY(None, is_view)
+cdef c_make_array(void* chandle, is_view, is_container):
+    # TODO(@junrushao1994): make it more like C...
+    global _TVM_ND_CLS
+    fcreate = _CLASS_NDARRAY
+    if is_container and len(_TVM_ND_CLS) > 0:
+        array_type_index = ctypes.cast(<unsigned long long>chandle, TVMNDArrayContainerHandle).array_type_index.value
```

Comment:
directly cast to a cython struct pointer, so you don't need to go through ctypes.